### ISSUE TITLE
[a11y]: SwitcherItem accessibility issues

### ISSUE THREAD

**Issue by andreikarasibm** [carbon-design-system/carbon#20424]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.90.0

### React version

19

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

There are several issues with the `Switcher` components:
1. When using a `SwitcherItem` component, it is possible that you don't give it an `href` attribute. You want it to behave as a button, but it always renders a link.
2. Components are not keyboard accessible by default. Once you give it the `tabIndex` param (which it should have by default since it is a link or a button), it becomes accessible, but you then have to make sure that it behaves correctly on keyboard navigation manually.
3. If you add `onKeyDown` prop, it will override the default behavior of a link, so you have to manually fire that link. For example, you have a tracking function you want to fire on a link click. The function you attach will override the default functionality of a link, so you have to manually trigger the link navigation.
4. If you give it `aria-label` param, it expects it to match the link text content exactly, which is not always the case.
5. Based on the configuration, you will encounter various a11y violations when checking it with IBM accessibility checker
6. When the component has several lines of text, it will be overflown by the next `SwitcherItem` element.

### Reproduction/example

https://stackblitz.com/edit/github-svvofpvt?file=src%2FApp.jsx

### Steps to reproduce

You can use the same code as in the stackblitz.
If to describe them in a list:

1. Add a few `SwitcherItem` components
2. Add a lot of text to the first one (will result in overflow)
3. Add href to any of them and check keyboard accessibillity
4. Add `onKeyDown` function to fire to a `SwitcherItem` component - it will not be keyboard accessible and will be rendered as a Link with no href
5. Add an `href` and a `onKeyDown` to a `SwitcherItem` component. Check, if it still works properly on keyboard navigation
6. Give some of the `SwitcherItem` components aria-labels and check for a11y violations.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
